shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java

```diff
@@ -157,53 +159,64 @@ public int run(CommandLine cl) throws AlluxioException, IOException {
     String[] args = cl.getArgs();
     int replication = FileSystemShellUtils.getIntArg(cl, REPLICATION_OPTION, DEFAULT_REPLICATION);
     Set<String> workerSet = new HashSet<>();
+    Set<String> excludeWorkerSet = new HashSet<>();
     Set<String> localityIds = new HashSet<>();
     if (cl.hasOption(HOST_FILE_OPTION.getLongOpt())) {
       String hostFile = cl.getOptionValue(HOST_FILE_OPTION.getLongOpt()).trim();
-      readLinesToSet(workerSet, hostFile);
+      readLinesToSet(workerSet, excludeWorkerSet, hostFile);
     } else if (cl.hasOption(HOSTS_OPTION.getLongOpt())) {
       String argOption = cl.getOptionValue(HOSTS_OPTION.getLongOpt()).trim();
-      readItemsFromOptionString(workerSet, argOption);
+      readItemsFromOptionString(workerSet, excludeWorkerSet, argOption);
     }
     if (cl.hasOption(LOCALITY_FILE_OPTION.getLongOpt())) {
       String localityFile = cl.getOptionValue(LOCALITY_FILE_OPTION.getLongOpt()).trim();
-      readLinesToSet(localityIds, localityFile);
+      readLinesToSet(localityIds, null, localityFile);
     } else if (cl.hasOption(LOCALITY_OPTION.getLongOpt())) {
       String argOption = cl.getOptionValue(LOCALITY_OPTION.getLongOpt()).trim();
-      readItemsFromOptionString(localityIds, argOption);
+      readItemsFromOptionString(localityIds, null, argOption);
```

Comment:
For the LOCALITY_OPTION, we can also exclude the `localityId` which starts with `!`